### ISSUE TITLE
Issues with `linkerd check --namespace [$namespace] --pre --linkerd-namespace --single-namespace` when execute with limited role.

### ISSUE THREAD

**Issue by jimcal** [linkerd/linkerd2#1846]
### What is the issue?
Because I am operating from an user within a enterprise version of kubernetes clusters, I only have namespace level permission. With the help from @rmars & @wmorgan, I was able to try out the `--single-namespace` feature in edge release. However, there seems to be sign of bugs and I will attempt to document them.  

### How can it be reproduced?
[the namespace I operate within is called `clickstream`]
I ran `linkerd check --namespace clickstream --pre --linkerd-namespace --single-namespace clickstream` 

### Logs, error output, etc

(If the output is long, please create a [gist](https://gist.github.com/) and
paste the link here.)

#### `linkerd check` output

```text
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
kubernetes-setup: control plane namespace does not already exist...........[FAIL] -- The "clickstream" namespace already exists
kubernetes-setup: can create Namespaces....................................[FAIL] -- Missing permissions to create Namespace

Status check results are [FAIL]
```

### Environment

- Kubernetes Version: Client Version: v1.9.6 / Server Version: v1.10.8
- Cluster Environment: Enterprise self administrated Kubernetes platform on AWS 
- Host OS: Container Linux by CoreOS 1855.4.0 (Rhyolite)
- Linkerd version: edge-18.11.1

**Comment by klingerf** [comment]
@jimcal Thanks for reporting this! I think there's an issue with the repro command that you pasted:

```
linkerd check --namespace clickstream --pre --linkerd-namespace --single-namespace clickstream
```

Shouldn't that be:

```
linkerd check --namespace clickstream --pre --linkerd-namespace clickstream --single-namespace
```

(`clickstream` is an argument to the `--linkerd-namespace` flag, not the `--single-namespace` flag; @wmorgan opened #1849 to track improving the interaction between these two flags)

But more to your point, I agree that if a user is attempting to install linkerd in single-namespace mode, it's likely that they don't have permissions to create arbitrary new namespaces, and instead the namespace to which they're trying to install has already been created for them. If that's the case, then we should modify the `linkerd check --single-namespace --pre` codepath to:

* Validate that the target namespace to which they plan to install linkerd _does_ exist
* Skip the permissions check to see if the user can create new namespaces

We can use this issue to track that work. Note that if we take this approach, we'll also need to modify the `linkerd install --single-namespace` output to omit the namespace object itself.

**Comment by jimcal** [comment]
@klingerf thanks for triaging this issue! 

I re-run the `linkerd check` command with your suggestion, and my output is still

```
| =>linkerd check --namespace clickstream --pre --linkerd-namespace clickstream --single-namespace
kubernetes-api: can initialize the client..................................[ok]
kubernetes-api: can query the Kubernetes API...............................[ok]
kubernetes-api: is running the minimum Kubernetes API version..............[ok]
kubernetes-setup: control plane namespace does not already exist...........[FAIL] -- The "clickstream" namespace already exists
kubernetes-setup: can create Namespaces....................................[FAIL] -- Missing permissions to create Namespace

Status check results are [FAIL]
```
I agree with your suggestion on how to solve this. Please let me know if I can be helpful in assisting further.

**Comment by klingerf** [comment]
This was fixed by #1974; closing.
